### PR TITLE
[MilkmApp] Add modal with price checker information

### PR DESCRIPTION
**Pull request by yvesfracari**
image

Token symbol is missing on the ponder, this is why is is blank.

image

### CODE REVIEW

**Comment by ribeirojose** [comment]
> image
> Token symbol is missing on the ponder, this is why is is blank.
> 
> image

Can we add that there @mendesfabio @luizakp ?

File: apps/cow-tools/src/lib/priceCheckers.ts [hunk 1]
```diff
@@ -205,3 +219,76 @@ function encodeWithExpectedOutCalculatorWithoutParameters(
...
+      return decodeArguments(expectedArgs, data);
+    }
+
+    return decodeWithExpectedOutCalculator(expectedArgs, data);
+  } catch {
+    return [];
+  }
+}
+
+export function getPriceCheckerInfoFromAddressAndChain(
```
Comment (ribeirojose) [inline apps/cow-tools/src/lib/priceCheckers.ts:240]: some comments explaining what these functions do/why would be helpful here

File: apps/cow-tools/src/app/milkman/(components)/OrdersTable.tsx [hunk 2]
```diff
@@ -109,14 +136,75 @@ function CancelButton({ status }: { status: string }) {
...
 function TokenInfo({ symbol, id }: { symbol?: string | null; id?: string }) {
   return (
     <div className="flex items-center gap-x-1">
       <div className="w-12">
         <div className="flex items-center justify-center">
           <div className="rounded-full bg-white p-1">
             <Image
-              src={"/assets/generic-token-logo.png"}
+              src={
+                tokenLogoUri[symbol as keyof typeof tokenLogoUri] ||
```
Comment (ribeirojose) [inline apps/cow-tools/src/app/milkman/(components)/OrdersTable.tsx:205]: isn't that info in the CoW tokens list?

File: apps/cow-tools/src/app/milkman/(components)/OrdersTable.tsx [hunk 3]
```diff
@@ -109,14 +136,75 @@ function CancelButton({ status }: { status: string }) {
...
+  const expecetedArguments = priceCheckerInfo?.arguments;
+
+  const decodedArgs = priceCheckerInfo
+    ? decodePriceCheckerData(
+        priceCheckerInfo,
+        order.priceCheckerData as `0x${string}`,
+      )
+    : [];
+
+  const priceCheckerUrl = buildBlockExplorerAddressURL({
```
Comment (ribeirojose) [inline apps/cow-tools/src/app/milkman/(components)/OrdersTable.tsx:154]:
```suggestion
  const priceCheckerExplorerUrl = buildBlockExplorerAddressURL({
```

File: apps/cow-tools/src/app/milkman/(components)/OrdersTable.tsx [hunk 4]
```diff
@@ -109,14 +136,75 @@ function CancelButton({ status }: { status: string }) {
...
+            <div key={index}>
+              {expecetedArguments[index].label} :{" "}
+              {expecetedArguments[index].convertOutput(
+                argument,
+                order.tokenOut?.decimals || 18,
+              )}
+            </div>
+          ))}
+        {!decodedArgs.length && (
+          <span>Price Checker Data: Error decoding data</span>
```
Comment (ribeirojose) [inline apps/cow-tools/src/app/milkman/(components)/OrdersTable.tsx:190]: inline logic like this makes the code a bit harder to parse -- would prefer to have separate functions / components when we have those scenarios to make this more friendly to review

**Review by ribeirojose — APPROVED** [review]

**Comment by luizakp** [comment]
> > image
> > Token symbol is missing on the ponder, this is why is is blank.
> > image
> 
> Can we add that there @mendesfabio @luizakp ?

Do you mean the token symbol? @ribeirojose

**Comment by luizakp** [comment]
image

decoding didn't work with the preview deploy

File: apps/cow-tools/src/app/milkman/(components)/OrdersTable.tsx [hunk 5]
```diff
@@ -109,14 +136,75 @@ function CancelButton({ status }: { status: string }) {
...
+    order.chainId as 5,
+    order.priceChecker as Address,
+  );
+
+  const expecetedArguments = priceCheckerInfo?.arguments;
+
+  const decodedArgs = priceCheckerInfo
+    ? decodePriceCheckerData(
+        priceCheckerInfo,
+        order.priceCheckerData as `0x${string}`,
```
Comment (luizakp) [inline apps/cow-tools/src/app/milkman/(components)/OrdersTable.tsx:150]:
```suggestion
        order.priceCheckerData as Address,
```

File: apps/cow-tools/src/app/milkman/(components)/OrdersTable.tsx [hunk 6]
```diff
@@ -109,14 +136,75 @@ function CancelButton({ status }: { status: string }) {
...
+  const priceCheckerUrl = buildBlockExplorerAddressURL({
+    chainId: order.chainId,
+    address: order.priceChecker as Address,
+  });
+
+  return (
+    <div className="text-white">
+      <div className="font-semibold text-2xl flex justify-between">
+        Price Checker Data - {truncateAddress(order.tokenIn?.symbol)} for{" "}
+        {truncateAddress(order.tokenOut?.symbol)}
```
Comment (luizakp) [inline apps/cow-tools/src/app/milkman/(components)/OrdersTable.tsx:163]:
```suggestion
        Price Checker Data - {order.tokenIn?.symbol} for{" "}
        {order.tokenOut?.symbol}
```

**Comment by yvesfracari** [comment]
@luizakp the error that you checked is because that price checker was created with invalid price checker data. So, that is why we are rendering an error there. If you check other orders, it will work 😄

File: apps/cow-tools/src/app/milkman/(components)/OrdersTable.tsx (hunk 5 above)
Comment (yvesfracari) [reply apps/cow-tools/src/app/milkman/(components)/OrdersTable.tsx:150]: Even that for us is the same thing this `priceCheckerData` is in bytes format, not an Address. I think that is more readable to let it is. wdyt?

**Pull request merged**